great_expectations/data_context/data_context.py

```diff
@@ -3264,6 +3270,41 @@ def run_checkpoint(
             **kwargs,
         )
 
+    def add_profiler(
+        self,
+        name: str,
+        config_version: float,
+        rules: Dict[str, dict],
+        class_name: str,
+        module_name: Optional[str] = None,
+        variables: Optional[dict] = None,
+        commented_map: Optional[CommentedMap] = None,
+        ge_cloud_id: Optional[str] = None,
+    ):
+        config_data = {
+            "name": name,
+            "config_version": config_version,
+            "rules": rules,
+            "class_name": class_name,
+            "module_name": module_name,
+            "variables": variables,
+            "commented_map": commented_map,
+        }
+        # Package config and perform validation
+        errors: Dict[str, List[str]] = ruleBasedProfilerConfigSchema.validate(
+            config_data
+        )
+        if errors:
+            raise ValidationError(
+                f"Something went wrong during profiler validation: {errors}"
+            )
+        config: RuleBasedProfilerConfig = RuleBasedProfilerConfig(**config_data)
```

Comment:
This part is up to debate but I think packaging a config object and having the RBP util method work that single object is much cleaner than accepting each single argument.

@alexsherstinsky let's also discuss `validate` and what makes sense here.

This method requires a test - this PR is to be blocked until that path is under test.

Reply:
@cdkini My preference would be for now to stick to the battle-tested "deserialize/serialize" (load/dump) pattern.  We are going to do a lot of rearchitecting of configuration and serialization, and I would feel more comfortable if we deferred this validation to that time as well (it will allow us to learn more about the applicability of each; moreover, it will be a great opportunity to consider upgrading to the latest "shipping" Marshmallow, instead of relying on our own patched rendition of an older version -- the latest release may give us even better validation tools).

Reply:
That's fair - I'll revert to the old pattern but I think we need to have two next steps:
* Determine if we can remove Marshmallow shade and work towards implementing the true library
* Abstract this `load/dump` pattern into a Schema-level function to provide some more context.